### DISCUSSION TITLE
SEQ Log4net configuration

### DISCUSSION THREAD

**Discussion by Abhishekkb18** [datalust/seq-tickets#1830] [Help] [answered]
Hello,

For the existing Log4net I'm trying to add Seq logging but unfortunately, out of 5 different projects in the solution it is working only for one project and I'm unable to figure out why is it working only for that particular project as the Log4net configs are similar for all the projects. I tried to run them as individual projects but no luck. Below are the steps I tried so far,

Step 1- Added Seq client and Log4net Async from Package Manager console
  PM> Install-Package Seq.Client.Log4Net
  PM> Install-Package Log4Net.Async

Step 2- Added Seq appenders to existing Log4net config (These are the changes made to my Log4net config)

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
	<configSections>
		<section name="log4net" type="log4net.Config.Log4NetConfigurationSectionHandler, log4net" />
	</configSections>
	<log4net>
My Log4net configs here
<appender name="SeqAppenderMaintenance" type="Seq.Client.Log4Net.SeqAppender, Seq.Client.Log4Net">
			<serverUrl value="http://localhost:5342/"/>
			<apiKey value=""/>
		</appender>
		<appender name="SeqAsyncForwarderMaintenance" type="Log4Net.Async.AsyncForwardingAppender,Log4Net.Async">
			<appender-ref ref="SeqAppenderMaintenance"/>
		</appender>
		<root>
			<level value="INFO"/>
			<appender-ref ref="SeqAsyncForwarderMaintenance"/>			
		</root>
	</log4net>
</configuration>
```

Step 3 - Installed Seq with the default configuration

Please let me know if I'm missing something or if I need to do anything on the project level for Seq to work. Thank you!

**Comment by nblumhardt** [comment]
Hi!

RE:

```xml
<serverUrl value="http://localhost:5342/"/>
```

The default port for Seq is 5341.

**Reply by Abhishekkb18** [reply]
Hi,
It is not working on any port. That is not the issue as I'm configuring Seq on that port.

**Reply by nblumhardt** [reply]
Hi Abishek,

Two things come to mind; first, if they're apps that start and stop, log4net may need some time to flush events before it's shut down. The log4net documentation would be the best place to check the requirements for this.

The second thing - have you checked the Seq ingestion log? There are some instructions in:

https://docs.datalust.co/docs/ingestion-troubleshooting

(You can ignore the Serilog section there.)

If neither of these things help, cutting down the code in one of the non-working projects until it's exactly identical to what you have in the working project should pinpoint the issue.

Hope this helps,
Nick

**Reply by Abhishekkb18** [reply, marked as answer]
Thanks For the reply, I was able to solve this by adding the appropriate buffer size in Seq appender.
`<appender name="SeqAppenderNotification" type="Seq.Client.Log4Net.SeqAppender, Seq.Client.Log4Net">
			<bufferSize value="1" />
			<serverUrl value="http://localhost:5341/"/>
			<apiKey value=""/>
		</appender>` 

By default it takes buffer size as 512.

**Reply by todthomson** [reply]
@Abhishekkb18 I’m presuming that buffer size 512 would mean we need to log 512 messages before those are sent to Seq.

I think it’s worth calling out here (in case anyone else stumbles on to this via a Google search) that `<bufferSize value="1" />` is the suggested value as per the getting started guide here: https://github.com/datalust/seq-client-log4net#getting-started.
